File: backend/API/serializers/audio.py
from rest_framework import serializers
from ..models import Audio
from . import BaseSerializer


class AudioSerializer(BaseSerializer):
    """
    Sérializer du node Audio

    Champs aditionnels :
        - name : nom de l'audio (obligatoire)
        Relations :
            - extraits : les extraits liés à l'audio
    """

    name = serializers.CharField(required=False, allow_blank=True, allow_null=True)

    # Outputs
    extraits = serializers.SerializerMethodField(read_only=True)

    def __init__(self, *args, **kwargs):
        """
        node : Audio
        """
        super().__init__(Audio, *args, **kwargs)

    def get_extraits(self, audio):
        """
        Renvoie un lien propre vers les extraits
        """
        return self.get_url("extrait-list", kwargs={"audio_uuid": audio.uuid})

File: frontend/src/model/audio.js
import Model from "@model/model.js";
import Extrait from "@model/extrait.js";

export default class Audio extends Model {
    #name;
    #extraits;

    constructor({ uuid, name, extraits }) {
        super(uuid);
        this.#name = name;
        this.#extraits = extraits;
    }

    static get endpoint() { return "audios"; }

    get name() { return this.#name; }
    set name(value) { this.#name = this.validateString(value, "name"); }

    async extraits(args) { return await this.fetchList(this.#extraits, Extrait, args); }

    fromJSON(json) {
        super.fromJSON(json);
        this.#name = json.name;
        this.#extraits = json.extraits;
        return this;
    }

    toJSON(json = {}) {
        json = super.toJSON(json);
        if (this.name) json['name'] = this.#name;
        return json;
    }
}